src/client/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -46,9 +49,19 @@ public bool CanInvokeBroker(CoreUIParent uiParent)
 
             uiParent.CallerViewController.InvokeOnMainThread(() =>
             {
-                result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl(BrokerParameter.BrokerV2));
+                result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl(BrokerParameter.BrokerV3));
+                if (result)
+                {
+                    BrokerRequestNonce = Guid.NewGuid().ToString();
+                    _logger.Info(iOSBrokerConstants.iOSBrokerv3Installed + result);
+                }
             });
 
+            if (!result)
+            {
+                result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl(BrokerParameter.BrokerV2));
```

Comment:
Why can this be on a non-UI thread but the call at line 52 must be on the UI thread?